### PR TITLE
protect tracker copy properly to avoid race

### PR DESCRIPTION
**Pull request by harshavardhana**
## Community Contribution License
All community contributions in this pull request are licensed to the project maintainers
under the terms of the [Apache 2 license](https://www.apache.org/licenses/LICENSE-2.0). 
By creating this pull request I represent that I have the right to license the 
contributions to the project maintainers under the Apache 2 license.

## Description
protect tracker copy properly to avoid race

## Motivation and Context
```
WARNING: DATA RACE
Write at 0x00c000aac1e0 by goroutine 1133:
  github.com/minio/minio/cmd.(*healingTracker).updateProgress()
      github.com/minio/minio/cmd/background-newdisks-heal-ops.go:183 +0x117
  github.com/minio/minio/cmd.(*erasureObjects).healErasureSet.func5()
      github.com/minio/minio/cmd/global-heal.go:292 +0x1d3

Previous read at 0x00c000aac1e0 by goroutine 1003:
  github.com/minio/minio/cmd.(*allHealState).updateHealStatus()
      github.com/minio/minio/cmd/admin-heal-ops.go:136 +0xcb
  github.com/minio/minio/cmd.(*healingTracker).save()
      github.com/minio/minio/cmd/background-newdisks-heal-ops.go:223 +0x424
```

## How to test this PR?
CI/CD tests this already.

## Types of changes
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Optimization (provides speedup with no functional changes)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
- [ ] Fixes a regression (If yes, please add `commit-id` or `PR #` here)
- [ ] Unit tests added/updated
- [ ] Internal documentation updated
- [ ] Create a documentation update request [here](https://github.com/minio/docs/issues/new?label=doc-change,title=Doc+Updated+Needed+For+PR+github.com%2fminio%2fminio%2fpull%2fNNNNN)

### CODE REVIEW

File: cmd/admin-heal-ops.go [hunk 1]
```diff
@@ -133,7 +133,10 @@ func (ahs *allHealState) popHealLocalDisks(healLocalDisks ...Endpoint) {
 func (ahs *allHealState) updateHealStatus(tracker *healingTracker) {
 	ahs.Lock()
 	defer ahs.Unlock()
+
+	tracker.mu.RLock()
```
Comment (klauspost) [inline cmd/admin-heal-ops.go:137]: Ah.. Since it is dereferenced by `*tracker`. Good spot!

**Review by klauspost — APPROVED** [review]

File: cmd/admin-heal-ops.go [hunk 2]
```diff
@@ -133,7 +133,10 @@ func (ahs *allHealState) popHealLocalDisks(healLocalDisks ...Endpoint) {
...
 	ahs.healStatus[tracker.ID] = *tracker
+	tracker.mu.Unlock()
```
Comment (klauspost) [inline cmd/admin-heal-ops.go]:
```suggestion
	tracker.mu.RUnlock()
```

**Review by klauspost — CHANGES_REQUESTED** [review]

**Review by klauspost — APPROVED** [review]

File: cmd/admin-heal-ops.go (hunk 2 above)
Comment (klauspost) [inline cmd/admin-heal-ops.go]:
```suggestion

	tracker.mu.RLock()
	t := *tracker
	t.QueuedBuckets = append(make([]string, 0, len(tracker.QueuedBuckets)), tracker.QueuedBuckets...)
	t.HealedBuckets = append(make([]string, 0, len(tracker.HealedBuckets)), tracker.HealedBuckets...)
	ahs.healStatus[tracker.ID] = t
	tracker.mu.RUnlock()
```

**Review by klauspost — CHANGES_REQUESTED** [review]
We are only shallow copying slices. Clone them.

File: cmd/admin-heal-ops.go [hunk 3]
```diff
@@ -133,7 +133,13 @@ func (ahs *allHealState) popHealLocalDisks(healLocalDisks ...Endpoint) {
 func (ahs *allHealState) updateHealStatus(tracker *healingTracker) {
 	ahs.Lock()
 	defer ahs.Unlock()
-	ahs.healStatus[tracker.ID] = *tracker
+
+	tracker.mu.RLock()
+	t := *tracker
```
Comment (vadmeste) [inline cmd/admin-heal-ops.go:138]: maybe we can add Clone() function and use it here

**Pull request merged**